fix(post-tool-verifier): detectWriteFailure false-positive on Edit success

## Bug

In `post-tool-verifier.mjs`, the `Edit` case emits
"Edit operation failed. Verify file exists and content matches exactly."
even when the edit actually succeeds.

## Reproduction

- Environment: `dangerouslyDisableSandbox: true`
- Edit tool return value: "The file has been updated successfully."
- PostToolUse hook output: "Edit operation failed. Verify file exists and content matches exactly."
- Manual `grep` confirms the change was applied correctly

## Root Cause Hypothesis

```js
// post-tool-verifier.mjs v4.13.5 — Edit case
if (structuredWriteFailure ||
    (!structuredWriteSuccess &&
     !isClaudeCodeWriteSuccess(toolOutput) &&
     detectWriteFailure(toolOutput))) {
  message = 'Edit operation failed. Verify file exists and content matches exactly.';
}
```

`isClaudeCodeWriteSuccess(toolOutput)` or `detectWriteFailure(toolOutput)` appears to
misclassify the success string "The file has been updated successfully." as a failure,
triggering the false-positive message.

## Impact

- Claude performs an unnecessary `grep` verification call after every successful Edit
- Repeated false-positive messages introduce noise into Claude's decision loop

## Versions Affected

Confirmed in v4.13.4 and v4.13.5 (identical code path in both)

Opened PR: https://github.com/Yeachan-Heo/oh-my-claudecode/pull/2877

Verification completed:
- \
> oh-my-claude-sisyphus@4.13.5 test
> vitest src/__tests__/post-tool-verifier.test.mjs


 RUN  v4.0.18 /home/bellman/Workspace/omc-issue-2876-post-tool-edit-success

 ✓ src/__tests__/post-tool-verifier.test.mjs (95 tests) 2755ms

 Test Files  1 passed (1)
      Tests  95 passed (95)
   Start at  06:00:57
   Duration  3.05s (transform 146ms, setup 0ms, import 180ms, tests 2.75s, environment 0ms) — 95 passed
- \
> oh-my-claude-sisyphus@4.13.5 lint
> eslint src


/home/bellman/Workspace/omc-issue-2876-post-tool-edit-success/src/__tests__/context-bloat-2577.test.ts
  17:10  warning  'existsSync' is defined but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars

/home/bellman/Workspace/omc-issue-2876-post-tool-edit-success/src/__tests__/rate-limit-wait/pane-fresh-capture.test.ts
  32:3  warning  Unused eslint-disable directive (no problems were reported from '@typescript-eslint/no-explicit-any')

/home/bellman/Workspace/omc-issue-2876-post-tool-edit-success/src/__tests__/setup-contracts-regression.test.ts
  15:37  warning  'statSync' is defined but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars

/home/bellman/Workspace/omc-issue-2876-post-tool-edit-success/src/hooks/auto-slash-command/executor.ts
  21:10  warning  'formatOmcCliInvocation' is defined but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars

/home/bellman/Workspace/omc-issue-2876-post-tool-edit-success/src/hooks/team-dispatch-hook.ts
  358:16  warning  'runProcess' is defined but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars

/home/bellman/Workspace/omc-issue-2876-post-tool-edit-success/src/installer/__tests__/stale-cleanup.test.ts
  12:75  warning  'readdirSync' is defined but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars

/home/bellman/Workspace/omc-issue-2876-post-tool-edit-success/src/team/__tests__/auto-merge.perf.test.ts
  10:10  warning  'existsSync' is defined but never used. Allowed unused vars must match /^_/u                @typescript-eslint/no-unused-vars
  10:22  warning  'readFileSync' is defined but never used. Allowed unused vars must match /^_/u              @typescript-eslint/no-unused-vars
  11:10  warning  'join' is defined but never used. Allowed unused vars must match /^_/u                      @typescript-eslint/no-unused-vars
  16:3   warning  'readEventLog' is defined but never used. Allowed unused vars must match /^_/u              @typescript-eslint/no-unused-vars
  77:13  warning  'FILES_PER_MERGE' is assigned a value but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars

/home/bellman/Workspace/omc-issue-2876-post-tool-edit-success/src/team/__tests__/merge-orchestrator.test.ts
    8:54  warning  'writeFileSync' is defined but never used. Allowed unused vars must match /^_/u    @typescript-eslint/no-unused-vars
  399:11  warning  'bCount' is assigned a value but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars

/home/bellman/Workspace/omc-issue-2876-post-tool-edit-success/src/team/__tests__/resolved-routing-snapshot.test.ts
  21:5  warning  'savedEnv' is never reassigned. Use 'const' instead  prefer-const

/home/bellman/Workspace/omc-issue-2876-post-tool-edit-success/src/team/__tests__/stage-router.test.ts
  23:5  warning  'savedEnv' is never reassigned. Use 'const' instead  prefer-const

/home/bellman/Workspace/omc-issue-2876-post-tool-edit-success/src/team/__tests__/teardown-invariant.test.ts
   7:36  warning  'mkdirSync' is defined but never used. Allowed unused vars must match /^_/u      @typescript-eslint/no-unused-vars
   7:47  warning  'writeFileSync' is defined but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars
  21:10  warning  'sanitizeName' is defined but never used. Allowed unused vars must match /^_/u   @typescript-eslint/no-unused-vars

/home/bellman/Workspace/omc-issue-2876-post-tool-edit-success/src/team/__tests__/worktree-runtime-e2e.test.ts
  7:22  warning  'readFileSync' is defined but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars

✖ 19 problems (0 errors, 19 warnings)
  0 errors and 3 warnings potentially fixable with the `--fix` option. — passed with existing warnings only (0 errors)
- \
> oh-my-claude-sisyphus@4.13.5 build
> tsc && node scripts/build-skill-bridge.mjs && node scripts/build-mcp-server.mjs && node scripts/build-bridge-entry.mjs && npm run compose-docs && npm run build:runtime-cli && npm run build:team-server && npm run build:cli

Built dist/hooks/skill-bridge.cjs
Built bridge/mcp-server.cjs
Built bridge/team-bridge.cjs

> oh-my-claude-sisyphus@4.13.5 compose-docs
> node scripts/compose-docs.mjs

Synced 0 partials to shared/
Documentation composition complete.

> oh-my-claude-sisyphus@4.13.5 build:runtime-cli
> node scripts/build-runtime-cli.mjs

Built bridge/runtime-cli.cjs

> oh-my-claude-sisyphus@4.13.5 build:team-server
> node scripts/build-team-server.mjs

Built bridge/team-mcp.cjs

> oh-my-claude-sisyphus@4.13.5 build:cli
> node scripts/build-cli.mjs

Built bridge/cli.cjs
Built bridge/team.js — passed
- \ — passed

--- a/scripts/post-tool-verifier.mjs
+++ b/scripts/post-tool-verifier.mjs
@@ -710,6 +710,7 @@ export function isClaudeCodeWriteSuccess(output) {
 
   const cleaned = stripClaudeTempCwdErrors(output);
   const successPatterns = [
+    /(^|\n)The file has been updated successfully\.?(\n|$)/i,
     /(^|\n)The file .+ has been updated successfully\.?(\n|$)/i,
     /(^|\n)File (?:created|written|updated) successfully(?:\s+at:\s*.+)?\.?(\n|$)/i,
     /(^|\n).*file state is current in your context\b.*(\n|$)/i,

--- a/src/__tests__/post-tool-verifier.test.mjs
+++ b/src/__tests__/post-tool-verifier.test.mjs
@@ -530,6 +530,35 @@ describe('post-tool hook regression coverage (issue #2615)', () => {
     expect(out.hookSpecificOutput?.additionalContext).not.toContain('Edit operation failed');
   });
 
+  it('prefers exact Claude Code edit success output over embedded diagnostics', () => {
+    const out = runPostToolVerifier({
+      tool_name: 'Edit',
+      tool_response: [
+        'The file has been updated successfully.',
+        '',
+        'diagnostic fixture: Error: failed to write',
+        'No such file or directory',
+      ].join('\n'),
+      session_id: 'issue-2876-exact-edit-success',
+      cwd: process.cwd(),
+    });
+
+    expect(isClaudeCodeWriteSuccess('The file has been updated successfully.')).toBe(true);
+    expect(out.hookSpecificOutput?.additionalContext).toContain('Code modified.');
+    expect(out.hookSpecificOutput?.additionalContext).not.toContain('Edit operation failed');
+  });
+
+  it('keeps exact edit failure text classified as an Edit failure', () => {
+    const out = runPostToolVerifier({
+      tool_name: 'Edit',
+      tool_response: 'Error: failed to edit file',
+      session_id: 'issue-2876-edit-failure',
+      cwd: process.cwd(),
+    });
+
+    expect(out.hookSpecificOutput?.additionalContext).toContain('Edit operation failed');
+  });
+
   it('prefers canonical write success output over serialized tool output with diagnostics', () => {
     const out = runPostToolVerifier({
       tool_name: 'Write',

